### PR TITLE
Task 66 side menu pr

### PR DESCRIPTION
**Pull request by alonkeyval**
This pull request introduces a new side menu component responsible for seamless navigation between pages. The side menu enhances the user experience by providing easy access to various sections of the application.

### CODE REVIEW

File: frontend/webapp/app/layout.tsx [hunk 1]
```diff
@@ -8,6 +8,8 @@ const LAYOUT_STYLE: React.CSSProperties = {
   margin: 0,
   position: "fixed",
   scrollbarWidth: "none",
+  width: "100vw",
+  height: "100vh",
```
Comment (blumamir) [inline frontend/webapp/app/layout.tsx:12]: Here you use `100vw` / `100vh` but below you use `100%`. is it intentional?

File: frontend/webapp/app/page.tsx [hunk 2]
```diff
@@ -24,11 +24,10 @@ export default function App() {
     switch (installation) {
       case CONFIG.NEW:
       case CONFIG.APPS_SELECTED:
-        router.push(`${ROUTES.SETUP}${state}`);
+        router.push(ROUTES.SETUP);
```
Comment (blumamir) [inline frontend/webapp/app/page.tsx:27]: Can you explain why the change? Looks like `state` above is currently unused and can be removed

File: frontend/webapp/components/side.menu/menu.item/menu.item.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    focused ? theme.colors.blue_grey : "transparent"};
+`;
+
+export default function MenuItem({ item, focused, onClick }: MenuItemProps) {
+  const { name, icons } = item;
+
+  return (
+    <MenuItemContainer onClick={onClick} focused={focused}>
+      {focused ? icons.focus() : icons.notFocus()}
+      <KeyvalText size={14} weight={600}>
```
Comment (blumamir) [inline frontend/webapp/components/side.menu/menu.item/menu.item.tsx:40]: Consider extracting these into a styled component so it's not hard coded in the component

File: frontend/webapp/components/side.menu/menu/items.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { OVERVIEW, ROUTES } from "@/utils/constants";
+import * as ICONS from "../../../assets/icons/side.menu";
+
+export const MENU_ITEMS = [
```
Comment (blumamir) [inline frontend/webapp/components/side.menu/menu/items.tsx:4]: Can we type it with the types defined above? That would guard us against type errors

Consider `interface MenuItem` from below

File: frontend/webapp/components/side.menu/menu/menu.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  function handleMenuItemClick(item) {
+    setCurrentMenuItem(item);
+    router.push(item?.navigate);
+  }
+
+  function renderMenuItemsList() {
+    return MENU_ITEMS.map((item) => (
+      <MenuItem
+        key={`${item.id}_${item.name}`}
```
Comment (blumamir) [inline frontend/webapp/components/side.menu/menu/menu.tsx:34]: Isn't `id` unique on it's own? do we need to add the name as well?

**Review by blumamir — APPROVED** [review]

File: frontend/webapp/app/layout.tsx (hunk 1 above)
Comment (alonkeyval) [reply frontend/webapp/app/layout.tsx:12]: The entire screen will occupy 100vw, the relative part of the page (e.g overview), will be 100% of the screen remaining after the width of the side menu, so will the other pages

File: frontend/webapp/app/page.tsx (hunk 2 above)
Comment (alonkeyval) [reply frontend/webapp/app/page.tsx:27]: the state should be remain, i will fix it

File: frontend/webapp/components/side.menu/menu.item/menu.item.tsx (hunk 3 above)
Comment (alonkeyval) [reply frontend/webapp/components/side.menu/menu.item/menu.item.tsx:40]: those are props sending to a reusable component
